### ISSUE TITLE
Moving out setup.py utilities

### ISSUE THREAD

**Issue by AbdealiJK** [davisking/dlib#119]
It would be awesome if you could move out your setup.py utilities into another pypi package so that other  devs can use it too !
i.e. if I have a package which gets compiled using cmake, I could just use that package (`setuptools-cmake` ?) and add some commands in the setup() to get a pypi package.

**Comment by davisking** [comment]
Yeah, that's a cool idea.  I was a little surprised at how big and complex the setup.py file needs to be.  It seems like there should be a much simpler interface like you suggest.  

I'm pretty busy with other stuff right now, but feel free to take this on yourself :)
